x-pack/plugins/actions/server/actions_config.ts

```diff
@@ -56,17 +56,17 @@ function disabledActionTypeErrorMessage(actionType: string) {
   });
 }
 
-function isAllowed({ allowedHosts }: ActionsConfigType, hostname: string): boolean {
+function isAllowed({ allowedHosts }: ActionsConfigType, hostname: string | null): boolean {
   const allowed = new Set(allowedHosts);
   if (allowed.has(AllowedHosts.Any)) return true;
-  if (allowed.has(hostname)) return true;
+  if (hostname && allowed.has(hostname)) return true;
   return false;
 }
 
 function isHostnameAllowedInUri(config: ActionsConfigType, uri: string): boolean {
   return pipe(
-    tryCatch(() => new URL(uri)),
-    map((url) => url.hostname),
+    tryCatch(() => url.parse(uri)),
```

Comment:
Updated to match how axios handles urls https://github.com/axios/axios/blob/59ab559386273a185be18857a12ab0305b753e50/lib/adapters/http.js#L91.